allow setting transparency color for tileset image collection

nobody-special666 opened the pull request:
This patch moves the transparency related widgets out of the Image group-box so it can be set when creating a image collection tileset. It also applies the transparency color when you add an image from the tileset dock.

![screenshot](https://cloud.githubusercontent.com/assets/901271/8150193/974d96b0-12ad-11e5-83b6-7659cd4261c3.png)
![screenshot2](https://cloud.githubusercontent.com/assets/901271/8150192/974b1c0a-12ad-11e5-886d-a4c8f92f505d.png)

bjorn commented:
Ah, I thought nobody used this feature anymore. ;-)

Anyway, good idea to still have it there since there is no reason not to support it. However, your patch is incomplete since the color would not only have to be applied in the Tilesets dock, but also when the map is loaded. I think the part that actually applies the color should be moved into the Tileset class.

nobody-special666 commented:
I fixed setting the transparent color when saving and loading a tileset and added the transparency color to the propertybrowser. Please review and commit

bjorn commented on src/libtiled/mapreader.cpp:
```diff
@@ -456,13 +465,6 @@ void MapReaderPrivate::readTilesetImage(Tileset *tileset)
-        if (!trans.startsWith(QLatin1Char('#')))
-            trans.prepend(QLatin1Char('#'));
-        tileset->setTransparentColor(QColor(trans));
-    }
```
Hmm, you can't just remove this code, since that breaks backwards compatibility. In general I'm not sure whether this change is really worth breaking compatibility at all. :-/

bjorn commented:
Maybe it could just be fixed while keeping compatibilty, by writing the `trans` attribute on each `image` element of the different tiles.

nobody-special666 commented:
I didn't just remove that code for transparency. I moved it out of readTilesetImage() where it only applies to loading a tileset image and moved it into readTileset() where it applies to both tileset image and collection. I tested it loading both types of tilesets and it works.

bjorn commented:
By moving it from `readTilesetImage`, which reads the `image` element inside a `tileset` element, to `readTileset`, which reads the `tileset` element, you're now expecting the `trans` attribute to be on the `tileset` element instead of the `image` element. This breaks compatibility.

nobody-special666 commented:
Solution: Then allow the property in both places and update the tileset transparency from both places. The front-end only allows 1 transparency color per tileset anyways so it should match at both levels.

nobody-special666 commented:
ok the trans property has been moved back to the image tag

nobody-special666 commented:
Is there still a problem preventing this commit from being merged?

bjorn commented:
Yes, the fact that I still need an almost full time job to make a living so it could take some days for me to get around to doing another review.
